.eslintrc.js

```diff
@@ -11,6 +14,46 @@ module.exports = {
     'src/govuk/vendor/polyfills/**/*'
   ],
   overrides: [
+    {
+      extends: 'plugin:jsdoc/recommended',
+      files: ['**/*.{cjs,js,mjs}'],
+      plugins: [
+        'jsdoc'
+      ],
+      rules: {
+        // JSDoc blocks are optional
+        'jsdoc/require-jsdoc': 'off',
+        'jsdoc/require-param-description': 'off',
+        'jsdoc/require-param': 'off',
+
+        // Require hyphens before param description
+        // Aligns with TSDoc style: https://tsdoc.org/pages/tags/param/
+        'jsdoc/require-hyphen-before-param-description': 'warn',
+
+        // Check for valid formatting
+        'jsdoc/check-line-alignment': 'warn',
+        'jsdoc/check-syntax': 'error',
+
+        // Add unknown @jest-environment tag name
+        'jsdoc/check-tag-names': [
+          'warn', {
+            definedTags: ['jest-environment']
+          }
+        ],
+
+        // Add missing .querySelectorAll() type
+        'jsdoc/no-undefined-types': [
+          'error', {
+            definedTypes: ['NodeListOf']
+          }
+        ]
+      },
+      settings: {
+        jsdoc: {
+          mode: 'typescript'
```

Comment:
Why `'typescript'` mode?

Rather than spend time documenting every complex object (or instance) we can use the declarations already published by the package author:

```mjs
/**
 * @param {import('puppeteer').Page} page
 * @param {import('puppeteer').ElementHandle} element
 * @param {import('puppeteer').WaitForOptions} options
 */
```

TypeScript declarations are widely available unlike local-only `@typedef` tags:

<img width="509" alt="Puppeteer on npmjs org showing TypeScript declarations are available" src="https://user-images.githubusercontent.com/415517/195338936-6aae0485-3c49-4bf7-9daf-dacd3fcb12c9.png">